torch/csrc/distributed/autograd/utils.cpp

```diff
@@ -153,7 +154,7 @@ std::shared_ptr<FutureMessage> sendMessageWithAutograd(
   std::shared_ptr<FutureMessage> fut;
   // If profiler is enabled, wrap this message with profiling metadata that will
   // tell the remote end to process this request with the profiler enabled.
-  if (torch::autograd::profiler::profilerEnabled()) {
+  if (!forceDisableProfiling && torch::autograd::profiler::profilerEnabled()) {
```

Comment:
This looks OK to me. But I wonder if we can avoid this new arg by making `torch::autograd::profiler::profilerEnabled()` return false? E.g., is it possible to use a guard in `rref_impl.cpp` to toggle the status of `profilerEnabled`?

Reply:
I guess one thing we could do is set a thread_local that allows this override, and have a guard that sets/restores it. If it's true, then we would override `profilerEnabled()` to always be false. Although, I don't think this should exist in `profiler::profilerEnabled()` itself since that may increase complexity in the profiler too much, maybe we can have a separate function for RPC profiling specifically that wraps around `profilerEnabled()` and checks this guard value.

Reply:
I see, if it's not that easy to toggle profiler on/off. The current version LGTM! Thanks!